src/topic/TopicMessageQuery.js

```diff
@@ -347,12 +347,19 @@ export default class TopicMessageQuery {
                         this._limit = this._limit.sub(1);
                     }
 
-                    this._startTime = Timestamp._fromProtobuf(
-                        /** @type {proto.ITimestamp} */ (
-                            message.consensusTimestamp
-                        )
+                    const consensusTimestamp = /** @type {proto.ITimestamp} */ (
+                        message.consensusTimestamp
                     );
 
+                    const nanos = /** @type {number} */ (
+                        consensusTimestamp.nanos
+                    );
+
+                    this._startTime = Timestamp._fromProtobuf({
+                        seconds: consensusTimestamp.seconds,
+                        nanos: nanos + 1,
+                    });
```

Comment:
This is quite a verbose way to increment a timestamp. But more importantly, as Xin noted to me it can overflow the nanos when nanos is `999,999,999` and not increment the seconds. Recommend you create a helper method in `Timestamp.plusNanos()` similar to how `java.time.Instant.plusNanos()` works that handles nanos larger than 9 digits.

```javascript
this._startTime = Timestamp._fromProtobuf((message.consensusTimestamp)).plusNanos(1)
```

Reply:
I think it's fair for the SDK to implement `Timestamp.plusNanos()`, but one questions comes to mind is how do the services work here? If I pass nanos > 999,999,999, does Hedera err, cap the nanos, or automatically combine nanos and seconds? Perhaps it shouldn't matter what the SDK does if the services already fix?

Reply:
HAPI assigns the consensus timestamp, so it will ensure it's with 0-999999999. It's not user input to HAPI. But it doesn't matter if the server validates it or not as this is not user supplied input but SDK calculated input. Since the SDK is wanting to increment by one nanos it should handle setting a correct value as per the contract that there's always at most 1B nanos in a second.